sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/OfferEvaluator.java

```diff
@@ -84,51 +123,24 @@ public OfferEvaluator(StateStore stateStore, OfferRequirementProvider offerRequi
+            Map<String, Protos.TaskInfo> thisPodTasks,
+            Optional<Protos.ExecutorInfo> executorInfo) {
         List<OfferEvaluationStage> evaluationPipeline = new ArrayList<>();
-
-        evaluationPipeline.add(new PlacementRuleEvaluationStage(stateStore.fetchTasks()));
-        if (offerRequirement.getExecutorRequirementOptional().isPresent()) {
-            ExecutorRequirement executorRequirement = offerRequirement.getExecutorRequirementOptional().get();
-
-            // We don't want to re-reserve resources for an already-running executor.
-            if (!executorRequirement.isRunningExecutor()) {
-                for (ResourceRequirement r : executorRequirement.getResourceRequirements()) {
-                    evaluationPipeline.add(r.getEvaluationStage(null));
-                }
-            }
-            evaluationPipeline.add(executorRequirement.getEvaluationStage());
-        } else {
-            evaluationPipeline.add(new ExecutorEvaluationStage());
-        }
-
-        for (TaskRequirement taskRequirement : offerRequirement.getTaskRequirements()) {
-            String taskName = taskRequirement.getTaskInfo().getName();
-            for (ResourceRequirement r : taskRequirement.getResourceRequirements()) {
-                evaluationPipeline.add(r.getEvaluationStage(taskName));
-            }
-
-            evaluationPipeline.add(taskRequirement.getEvaluationStage());
+        if (podInstanceRequirement.getPodInstance().getPod().getPlacementRule().isPresent()) {
+            evaluationPipeline.add(new PlacementRuleEvaluationStage(
+                    allTasks, podInstanceRequirement.getPodInstance().getPod().getPlacementRule().get()));
         }
-        evaluationPipeline.add(new ReservationEvaluationStage(offerRequirement.getResourceIds()));
-
-        return evaluationPipeline;
-    }
 
-    private static void logOutcome(StringBuilder stringBuilder, EvaluationOutcome outcome, String indent) {
-        stringBuilder.append(String.format("  %s%s%n", indent, outcome.toString()));
-        for (EvaluationOutcome child : outcome.getChildren()) {
-            logOutcome(stringBuilder, child, indent + "  ");
-        }
-    }
-
-    private OfferRequirement getOfferRequirement(PodInstanceRequirement podInstanceRequirement)
-            throws InvalidRequirementException {
         PodInstance podInstance = podInstanceRequirement.getPodInstance();
-        boolean noTasksExist = TaskUtils.getTaskNames(podInstance).stream()
-                .map(taskName -> stateStore.fetchTask(taskName))
-                .filter(taskInfoOptional -> taskInfoOptional.isPresent())
-                .map(taskInfoOptional -> taskInfoOptional.get())
+        boolean noLaunchedTasksExist = thisPodTasks.values().stream()
+                .flatMap(taskInfo -> taskInfo.getResourcesList().stream())
+                .map(resource -> ResourceCollectionUtils.getResourceId(resource))
+                .filter(resourceId -> resourceId.isPresent())
```

Comment:
could turn this filter+map+filter into a single `filter(resourceId -> resourceId.isPresent() && !resourceId.get().isEmpty())`